custom_components/econnect_metronet/binary_sensor.py

```diff
@@ -32,23 +32,23 @@ async def async_setup_entry(
     inventory = await hass.async_add_executor_job(device._connection._get_descriptions)
     for sector_id, name in inventory[query.SECTORS].items():
         unique_id = f"{entry.entry_id}_{DOMAIN}_{query.SECTORS}_{sector_id}"
-        sensors.append(EconnectDoorWindowSensor(unique_id, sector_id, entry, name, coordinator, device, query.SECTORS))
+        sensors.append(SectorSensor(unique_id, sector_id, entry, name, coordinator, device, query.SECTORS))
```

Comment:
Now that I think about it, I don't think you need to pass the sensor type anymore (you have multiple classes now and no `if` statements on that attribute). I guess you can remove `query.SECTORS` and `query.INPUTS` and their relative attribute and constructor parameter.